### PR TITLE
feat(VIST-CPC-927): implementing aggregation and details page

### PR DESCRIPTION
**Pull request by TheyCallMeCris**
JIRA:[ link to jira ticket](https://champlainsaintlambert.atlassian.net/browse/CPC-927)
## Context:
Visits only had access to the petId and vetId. By aggregating it, we now have access to the needed vet information and pet information. There was also no details page for our visits. The user only had access to the general view which is a list.
This also allows us to populate a details page with a single get method. 
## Changes

-Changed visitResponseDTO to now have the vet's full name and contact information as well as the pet's birth date and name.
-Added navigation to Details Page
-Added Details Page to the front-end

UI CHANGES:

**BEFORE:**

image

**AFTER:**

image

DETAILS PAGE:
image

### CODE REVIEW

File: visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/Utils/EntityDtoUtil.java [hunk 1]
```diff
@@ -2,30 +2,61 @@
...
-        BeanUtils.copyProperties(visit, visitResponseDTO);
-        return visitResponseDTO;
+
+        Mono<PetResponseDTO> petResponseDTOMono = petsClient.getPetById(visit.getPetId());
+        Mono<VetDTO> vetResponseDTOMono = vetsClient.getVetByVetId(visit.getPractitionerId());
+
+        return Mono.zip(petResponseDTOMono, vetResponseDTOMono)
+                .flatMap(tuple -> {
+                    PetResponseDTO petResponseDTO = tuple.getT1();
+                    VetDTO vetResponseDTO = tuple.getT2();
```
Comment (MichaelVacca) [inline visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/Utils/EntityDtoUtil.java:34]: Can you clarify this?

**Review by MichaelVacca — APPROVED** [review]
Looks good, great job implementing the aggregation!

File: api-gateway/src/main/resources/static/scripts/visit-list/visit-list.template.html [hunk 2]
```diff
@@ -100,11 +100,12 @@ <h3 style="margin: 0; display: inline-block;">Upcoming Visits</h3>
         <td class="border"></td>
         <td class="border"></td>
         <td class="border"></td>
+        <td class="border"></td>
```
Comment (swafit) [inline api-gateway/src/main/resources/static/scripts/visit-list/visit-list.template.html:103]: Thank you for fixing the missing td on the table.

File: visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/Utils/EntityDtoUtil.java (hunk 1 above)
Comment (TheyCallMeCris) [reply visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/Utils/EntityDtoUtil.java:34]: Mono.zip produces a tuple which can vary from type. In this case, there is Mono<PetResponseDTO> as the first item and Mono<VetDTO> as the second. I am then unzipping those items and passing the information that is inside of them to items of their respective types.

File: visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/VisitsServiceNewApplication.java [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.ClassPathResource;
```
Comment (swafit) [inline visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/VisitsServiceNewApplication.java:6]: Are these importd necessary?
```suggestion
```

File: visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/VisitsServiceNewApplication.java [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
```
Comment (swafit) [inline visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/VisitsServiceNewApplication.java:5]:
```suggestion
```

**Review by MichaelVacca — APPROVED** [review]

**Review by swafit — APPROVED** [review]
Fantastic work with the aggregation and the testing.

**Pull request merged**